tockloader-lib/src/attributes/get_app_attributes.rs

```diff
@@ -0,0 +1,146 @@
+
+        let app_version: u32 = LittleEndian::read_u32(&app_header_data[52..56]);
+
+        let tvl_package_type = LittleEndian::read_u16(&app_header_data[56..58]);
+
+        let package_length = LittleEndian::read_u16(&app_header_data[58..60]);
+
+        let package_name = bytes_to_string(&app_header_data[60..(60 + package_length as usize)]);
+
+        let tvl_kernel_version_type = LittleEndian::read_u16(
+            &app_header_data[(header_size as usize - 6)..(header_size as usize - 4)],
+        );
+
+        let kernel_major = LittleEndian::read_u16(
+            &app_header_data[(header_size as usize - 4)..(header_size as usize - 2)],
+        );
+
+        let kernel_minor = LittleEndian::read_u16(
+            &app_header_data[(header_size as usize - 2)..header_size as usize],
+        );
+
+        let kernel_version = (kernel_major, kernel_minor);
+
+        let footer_size = total_size - binary_end_offset;
+
+        details.tbf_version = Some(tbf_version);
+        details.header_size = Some(header_size);
+        details.total_size = Some(total_size);
+        details.checksum = Some(checksum);
+        details.flag_enabled = Some(enable);
+        details.flag_sticky = Some(sticky);
+        details.tvl_main_type = Some(tvl_main_type);
+        details.init_fn_offset = Some(init_fn_offset);
+        details.protected_size = Some(protected_size);
+        details.minumum_ram_size = Some(minimum_ram_size);
+        details.tvl_program_type = Some(tvl_program_type);
+        details.binary_end_offset = Some(binary_end_offset);
+        details.app_version = Some(app_version);
+        details.tvl_package_type = Some(tvl_package_type);
+        details.package_name = Some(package_name);
+        details.tvl_kernel_version_type = Some(tvl_kernel_version_type);
+        details.kernel_major = Some(kernel_major);
+        details.kernel_minor = Some(kernel_minor);
+        details.kernel_version = Some(kernel_version);
+        details.footer_size = Some(footer_size);
+
+        apps_details.insert(apps_counter, details);
+        apps_counter += 1;
+        address += total_size as u64;
+    }
+    apps_details
+}
+
+// TODO(MicuAna): add error handling
+pub(crate) fn get_start_address(board_core: &mut Core, mut address: u64) -> Option<u64> {
+    loop {
+        // Read a block of 200 8-bit words
+        let mut buff = vec![0u8; 200];
+        match board_core.read(address, &mut buff) {
+            Ok(_) => {}
+            Err(e) => {
+                println!("Error reading memory: {:?}", e);
+                break;
+            }
+        }
+        let (_ver, _header_len, whole_len) =
+            match parse_tbf_header_lengths(&buff[0..8].try_into().unwrap()) {
+                Ok((ver, header_len, whole_len)) if header_len != 0 => (ver, header_len, whole_len),
+                _ => break, // No more apps
+            };
+
+        address += whole_len as u64;
+    }
+    Some(address)
+}
```

Comment:
Is this function appropriate to put here? This doesn't seem relevant to `app_attributes`. Also, see comment attached to file